### PR TITLE
add sendable to support concurrency

### PR DESCRIPTION
**Pull request by angiemugo**
- Sendable is Swift's way of handling concurrency
- After enabling strict checking I noticed some places in the codebase that had this warning 
- Since a lot of the libraries we are using haven't enforced the `sendable` protocol, I silenced those warnings by adding `preconcurrency`. 
- We still have a potential concurrency error with how we handle `performInitialNavigationSetup`.
- Here's the reference for this https://developer.apple.com/documentation/swift/sendable

### CODE REVIEW

**Review by savekirk — APPROVED** [review]
LGTM

**Comment by cedricbahirwe** [comment]
Hey @angiemugo , what's the concurrency issue with @`performInitialNavigationSetup`?

File: Exercism/Util/NavigationModel.swift [hunk 1]
```diff
@@ -82,12 +82,12 @@ class NavigationModel: ObservableObject, Codable {
...
-        path.removeLast()
-    }
-    
     enum CodingKeys: String, CodingKey {
         case path
         case columnVisibility
     }
 }
+
+extension DispatchQueue {
```
Comment (cedricbahirwe) [inline Exercism/Util/NavigationModel.swift:91]: ???

Comment (angiemugo) [reply Exercism/Util/NavigationModel.swift:91]: My bad, I'll remove that

**Comment by angiemugo** [comment]
> Hey @angiemugo , what's the concurrency issue with @`performInitialNavigationSetup`?

`Instance methods of non-Sendable types cannot be marked as '@Sendable'; this is an error in Swift 6` we have this warning. `Navigation Model` type is unsafe and prone to data races. Off the top of my head, I would want to isolate it to its own actor and not expose `NavData` to be mutated the way we are right now.

**Pull request merged**